Fix bad DEVICE_TOKEN issue

aneilbaboo opened the pull request:
This fixes a problem due to a hard-coded device token in endpoints.py.  Vine is returning 502s if this value is set.  I changed the default to None, and provided an optional device_token keyword param in .login if the user wishes to set this explicitly.  

It works.

davoclavo commented:
I think they just blocked that particular `device_token`. In master I changed the token and requests work perfectly even if its hardcoded.

I like your solution better, but having the user provide one will make things more difficult. Why not generating a random one, like this:

```py
import os, binascii
binascii.b2a_hex(os.urandom(32))
```

aneilbaboo commented:
Good idea - I can make that change.

aneilbaboo commented:
By the way - is the device token even needed?  I was setting it to None and everything worked fine.

davoclavo commented:
Good point, it does work without a device token. Probably requests from their web client do not send a device token. I think as we are using the iPad/iPhone User-Agent header, it would be better to use a token.

Btw, remember to [sync your fork](https://help.github.com/articles/syncing-a-fork/) before doing the pull request so I can automatically merge it.

aneilbaboo commented:
Will do.  I have no idea why there were merge conflicts.  I forked just
before I made the changes.  Not more than 10 min elapsed between first
fork, commit and pull request.  Oh well, I'll fix it.

On Thu, Jan 22, 2015 at 3:08 PM, David <notifications@github.com> wrote:

> Good point, it does work without a device token. Probably requests from
> their web client do not send a device token. I think as we are using the
> iPad/iPhone User-Agent header, it would be better to use a token.
>
> Btw, remember to sync your fork
> <https://help.github.com/articles/syncing-a-fork/> before doing the pull
> request so I can automatically merge it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/davoclavo/vinepy/pull/10#issuecomment-71118122>.
>

aneilbaboo committed:
Fix bad DEVICE_TOKEN issue

By default, the device token is set to a random hex string each time
the library is loaded instead of using a hard-coded value.

aneilbaboo commented:
I made the fix you suggested.  A new device token is generated when the endpoints module is imported.  The user can override with a custom value.

davoclavo commented:
I think you committed the wrong changes, can't find the code where it generates the token. Also, some changes on master are being reverted by your commit :fearful:

davoclavo commented:
Thanks a lot for the proposal. I just added it and wrote some tests.

https://github.com/davoclavo/vinepy/blob/master/vinepy/api.py#L18

aneilbaboo commented:
Cool, thanks, David!

On Sun, Jan 25, 2015 at 5:21 PM, David <notifications@github.com> wrote:

> Closed #10 <https://github.com/davoclavo/vinepy/pull/10>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/davoclavo/vinepy/pull/10#event-224398132>.
>
